src/api_client.rs

```diff
@@ -1,560 +1,596 @@
-pub enum ConnectionOrientation {
-    Inbound,
-    Outbound,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct PeerDescriptor {
-    pub state: PeerState,
-    pub direction: ConnectionOrientation,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct PeerDescription {
-    pub peer_id: PeerId,
-    pub enr: Enr,
-    pub last_seen_p2p_address: Multiaddr,
-    pub state: PeerState,
-    pub direction: ConnectionOrientation,
-}
+#[tokio::main]
+pub async fn make_request(
+    ip_addr: &str,
+    endpoint: &str,
+    state_id: &str,
+    port_id: &str
+) -> Result<serde_json::Value, reqwest::Error> {
+    
+    let url: String = format!("http://{}:{}/{}/{}", ip_addr, port_id, endpoint, state_id);
 
-#[derive(Serialize, Deserialize)]
-pub struct PeerSummary {
-    #[serde(with = "crate::serde::as_string")]
-    pub disconnected: usize,
-    #[serde(with = "crate::serde::as_string")]
-    pub connecting: usize,
-    #[serde(with = "crate::serde::as_string")]
-    pub connected: usize,
-    #[serde(with = "crate::serde::as_string")]
-    pub disconnecting: usize,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct SyncStatus {
-    #[serde(with = "crate::serde::as_string")]
-    pub head_slot: Slot,
-    #[serde(with = "crate::serde::as_string")]
-    pub sync_distance: usize,
-    pub is_syncing: bool,
-}
-
-#[derive(Serialize, Deserialize)]
-pub enum HealthStatus {
-    Ready,
-    Syncing,
-    NotInitialized,
-}
+    println!("{:?}", &url);
 
-#[derive(Serialize, Deserialize)]
-pub struct AttestationDuty {
-    pub pubkey: BlsPublicKey,
-    #[serde(with = "crate::serde::as_string")]
-    pub validator_index: ValidatorIndex,
-    #[serde(with = "crate::serde::as_string")]
-    pub committee_index: CommitteeIndex,
-    #[serde(with = "crate::serde::as_string")]
-    pub committee_length: usize,
-    #[serde(with = "crate::serde::as_string")]
-    pub committees_at_slot: usize,
-    #[serde(with = "crate::serde::as_string")]
-    pub validator_committee_index: usize,
-    #[serde(with = "crate::serde::as_string")]
-    pub slot: Slot,
-}
+    let client = reqwest::Client::new();
 
-#[derive(Serialize, Deserialize)]
-pub struct ProposerDuty {
-    pub pubkey: BlsPublicKey,
-    #[serde(with = "crate::serde::as_string")]
-    pub validator_index: ValidatorIndex,
-    #[serde(with = "crate::serde::as_string")]
-    pub slot: Slot,
-}
+    let response = client.get(&url).send().await?;
 
-#[derive(Serialize, Deserialize)]
-pub struct SyncCommitteeDuty {
-    pub pubkey: BlsPublicKey,
-    #[serde(with = "crate::serde::as_string")]
-    pub validator_index: ValidatorIndex,
-    #[serde(with = "crate::serde::collection_over_string")]
-    pub validator_sync_committee_indices: Vec<usize>,
-}
-
-pub struct CommitteeDescriptor {
-    pub validator_index: ValidatorIndex,
-    pub committee_index: CommitteeIndex,
-    pub committees_at_slot: usize,
-    pub slot: Slot,
-    pub is_aggregator: bool,
-}
+    let body = response.text().await.unwrap();
```

Comment:
there is a built-in `json` method `reqwest` has to parse the response body as json

https://docs.rs/reqwest/latest/reqwest/#json